test/TabExpansion.Tests.ps1

```diff
@@ -0,0 +1,102 @@
+. $PSScriptRoot\Shared.ps1
+
+Describe 'TabExpansion Tests' {
+    Context 'Fetch/Push/Pull TabExpansion Tests' {
+        It 'Tab completes all remotes' {
+            $result = & $module GitTabExpansionInternal 'git push '
+            $result | Should BeExactly 'origin'
+        }
+        It 'Tab completes all branches' {
+            $result = & $module GitTabExpansionInternal 'git push origin '
+            $result -contains 'master' | Should Be $true
+            $result -contains 'origin/master' | Should Be $true
+            $result -contains 'origin/HEAD' | Should Be $true
+        }
+        It 'Tab completes all :branches' {
+            $result = & $module GitTabExpansionInternal 'git push origin :'
+            $result -contains ':master' | Should Be $true
+            # Is the following a valid branch name in a refspec?

```

Comment:
Thanks.  I wasn't sure about that but it didn't look right (with spaces, ->).  Thanks for updating the sample specs.  I "think" the multiple ref specs should be easy to implement/test.  I'll look at that tonight.